app/controllers/authentications_controller.rb

```diff
@@ -1,11 +1,22 @@
 class AuthenticationsController < ApplicationController
   def create
-    path = Authenticator.new(request.env['omniauth.auth'], session).perform
+    session[:return_to] ||= request.referer
+    Authenticator.new(request.env['omniauth.auth'], session).perform
     redirect_to(path)
   end
 
   def destroy
     reset_session
     redirect_to '/'
   end
+
+  private
+
+  def path
```

Comment:
Do you think we should just have this method do the redirect_to as well and name it redirect_to_logged_in?